src/purview.ts

```diff
@@ -326,11 +324,19 @@ export function handleWebSocket(
 
     ws.on("close", async () => {
       wsState.closing = true
-      if (!wsState.connected) {
+      // We check if `connectionState` is set to "connected" because it could be
+      // the case that the "close" event fires just after the "connect" event
+      // (e.g., on page refresh), and the "close" event will see that the
+      // `wsState.roots` is an empty array due to the "connect" still being in
```

Comment:
could we remove the quotes around "connect" or explicitly write `"connect" event`? it might be worthwhile to mention that the connect logic is async, and that's why it might be in progress